packages/material-ui/src/ButtonBase/ButtonBase.js

```diff
@@ -212,12 +219,12 @@ class ButtonBase extends React.Component {
     }
 
     // Fix for https://github.com/facebook/react/issues/7769
-    if (!this.button) {
-      this.button = event.currentTarget;
+    if (!this.buttonRef.current) {
```

Comment:
This branch is not covered anymore. The behavior was never tested anyway though. Researching the correct behavior that we would have to mock should be very expensive considering the age of the react issue. It essentially removes a false positive line coverage report.